lib/msf/core/rpc/v10/rpc_job_status_tracker.rb

```diff
@@ -7,66 +7,93 @@ class RpcJobStatusTracker
       include MonitorMixin
```

Comment:
Including the `MonitorMixin` isn't enough to have a thread safe object. You need to explicitly mark the synchronized areas, and ensure that `super` is called - as the behind the scenes the mixin initialises an internal mutex that it makes use of.